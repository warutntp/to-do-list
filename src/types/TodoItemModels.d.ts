export module TodoItemModels {
  export interface TodoItemModel {
    slug: string;
    title: string;
    description?: string;
    completed: boolean;
  }
}
