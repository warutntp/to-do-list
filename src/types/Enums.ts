export enum FilterTodoType {
  Completed = "completed",
  Incomplete = "incomplete",
  All = "all",
}
