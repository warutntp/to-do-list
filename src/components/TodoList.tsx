import React from "react";
import { TodoItemModels } from "../types/TodoItemModels";
import TodoItem from "./TodoItem";

interface TodoListProps {
  filteredTodos: TodoItemModels.TodoItemModel[];
}

const TodoList: React.FC<TodoListProps> = ({ filteredTodos }) => {
  return (
    <div className="table-responsive">
      <table className="table table-hover table-striped">
        <thead>
          <tr>
            <th scope="col" className="col-1 text-center">
              {" "}
              #{" "}
            </th>
            <th scope="col" className="col-3 text-wrap-break">
              Title
            </th>
            <th scope="col" className="col-4 text-wrap-break">
              Description
            </th>
            <th scope="col" className="col-2 text-center">
              Completed
            </th>
            <th scope="col" className="col-2 text-center">
              Action
            </th>
          </tr>
        </thead>
        {filteredTodos && filteredTodos.length > 0 ? (
          <tbody>
            {filteredTodos.map(
              (todo: TodoItemModels.TodoItemModel, index: number) => (
                <TodoItem key={todo.slug} todo={todo} index={index} />
              )
            )}
          </tbody>
        ) : (
          <tbody>
            <tr>
              <td colSpan={5} className="text-center">
                No todos found
              </td>
            </tr>
          </tbody>
        )}
      </table>
    </div>
  );
};

export default TodoList;
