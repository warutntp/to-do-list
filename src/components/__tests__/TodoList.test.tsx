import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import TodoList from "../TodoList";
import { TodoItemModels } from "../../types/TodoItemModels";

const mockStore = configureStore([]);
const store = mockStore({
  todos: [
    {
      slug: "1",
      title: "Test Todo",
      description: "Test description",
      completed: false,
    },
  ],
});

const todos: TodoItemModels.TodoItemModel[] = [
  {
    slug: "1",
    title: "Test Todo",
    description: "Test description",
    completed: false,
  },
];

describe("TodoList", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <TodoList filteredTodos={todos} />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should match snapshot no have todo", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <TodoList filteredTodos={[]} />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
