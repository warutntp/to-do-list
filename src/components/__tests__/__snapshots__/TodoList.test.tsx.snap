// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="table-responsive"
  >
    <table
      class="table table-hover table-striped"
    >
      <thead>
        <tr>
          <th
            class="col-1 text-center"
            scope="col"
          >
             # 
          </th>
          <th
            class="col-3 text-wrap-break"
            scope="col"
          >
            Title
          </th>
          <th
            class="col-4 text-wrap-break"
            scope="col"
          >
            Description
          </th>
          <th
            class="col-2 text-center"
            scope="col"
          >
            Completed
          </th>
          <th
            class="col-2 text-center"
            scope="col"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="col-1 text-center"
          >
            1
          </td>
          <td
            class="col-3 text-wrap-break"
          >
            Test Todo
          </td>
          <td
            class="col-4 text-wrap-break"
          >
            Test description
          </td>
          <td
            class="col-2"
          >
            <div
              class="form-check form-switch d-flex justify-content-center"
            >
              <input
                class="form-check-input"
                id="1"
                type="checkbox"
              />
            </div>
          </td>
          <td
            class="text-center"
          >
            <button
              class="btn py-0"
              title="Delete"
              type="button"
            >
              <i
                class="bi bi-trash bi-danger"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`TodoList should match snapshot no have todo 1`] = `
<DocumentFragment>
  <div
    class="table-responsive"
  >
    <table
      class="table table-hover table-striped"
    >
      <thead>
        <tr>
          <th
            class="col-1 text-center"
            scope="col"
          >
             # 
          </th>
          <th
            class="col-3 text-wrap-break"
            scope="col"
          >
            Title
          </th>
          <th
            class="col-4 text-wrap-break"
            scope="col"
          >
            Description
          </th>
          <th
            class="col-2 text-center"
            scope="col"
          >
            Completed
          </th>
          <th
            class="col-2 text-center"
            scope="col"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="text-center"
            colspan="5"
          >
            No todos found
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
